plugins/async-node/core/src/index.test.ts

```diff
@@ -119,6 +114,86 @@ test("should return current node view when the resolved node is undefined", asyn
   await asyncNodeTest(undefined);
 });
 
+test("should handle the promise using .then", async () => {
+  const plugin = new AsyncNodePlugin({
+    plugins: [new AsyncNodePluginPlugin()],
+  });
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  plugin.hooks.onAsyncNode.tap("test", async (node: Node.Node) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve;
+    });
+  });
+  let updateNumber = 0;
+
+  const player = new Player({ plugins: [plugin] });
+
+  player.hooks.viewController.tap("async-node-test", (vc) => {
+    vc.hooks.view.tap("async-node-test", (view) => {
+      view.hooks.onUpdate.tap("async-node-test", (update) => {
+        updateNumber++;
+      });
+    });
+  });
+
+  player.start(basicFRFWithActions as any);
+
+  let view = (player.getState() as InProgressState).controllers.view.currentView
+    ?.lastUpdate;
+
+  expect(view).toBeDefined();
+  expect(view?.actions[1]).toBeUndefined();
+
+  await waitFor(() => {
+    expect(updateNumber).toBe(1);
+    expect(deferredResolve).toBeDefined();
+  });
+
+  if (deferredResolve) {
+    deferredResolve([
+      {
+        asset: {
+          id: "value-1",
+          type: "text",
+          value: "1st value in the multinode",
+        },
+      },
+      {
+        id: "another-async",
+        async: true,
+      },
```

Comment:
For testing this use case, if you resolve with another async node, the `deferredResolve` value will be updated to be the next async node. We'll want to explicitly test this without triggering another async node resolution to ensure we use the same promise.

Reply:
Yes the reason I have explicitly resolved with another async node is because the same promise is not getting explicitly resolved . I wanted to show this behaviour to you so that we understand that for the given business logic this test actually fails . If you'd like to see that behaviour I can commit that code but it will not pass . I think we would need to revisit the logic in that case

Reply:
Okay, gotcha - yeah, let's get a test case in explicitly for what we're trying to solve for. If it fails, that's fine, we can revisit the implementation logic to get it passing 👍